Ginger/GingerCoreNETUnitTest/GingerRunnerTests/GingerRunnerTest.cs

```diff
@@ -464,5 +466,121 @@ public RunSetConfig CreteRunsetWithOperations()
+
+            //Act
+
+            mGR.RunAction(action);
+
+            //Assert
+            Assert.AreEqual(action.Status, Amdocs.Ginger.CoreNET.Execution.eRunStatus.Skipped);
+            Assert.AreEqual("Action is not active.", action.ExInfo);
+
+        }
+
+        [TestMethod]
+        [Timeout(60000)]
+        public void RunActionWithFlowControlAndMoveNext()
+        {
+            //Arrange
+            Activity activity = GetActivityFromRepository();
+
+            ActDummy act1 = new ActDummy();
+            act1.FileName = "FlowControlAction";
+            act1.Active = true;
+            activity.Acts.Add(act1);
+
+            ObservableList<IAct> actionList = activity.Acts;
+            Act action = (Act)actionList[0]; 
+            FlowControl flowControl = new FlowControl();
+            flowControl.Active = true;
+
+            flowControl.Operator = eFCOperator.ActionPassed;
+            flowControl.FlowControlAction = eFlowControlAction.GoToAction;
+            flowControl.Value= act1.Guid + flowControl.GUID_NAME_SEPERATOR + act1.ItemName;
+
+            action.FlowControls.Add(flowControl);
+
+            //Act
+            mGR.RunAction(action);
+
+            //Assert
+            Assert.AreEqual(action.Status, Amdocs.Ginger.CoreNET.Execution.eRunStatus.Passed);
+            Assert.AreSame(mGR.CurrentBusinessFlow.CurrentActivity.Acts.CurrentItem, act1);
+        }
+
+        [TestMethod]
+        [Timeout(60000)]
+        public void RunActionWithoutFlowControlAndMoveNext()
+        {
+            //Arrange
+            Activity activity = GetActivityFromRepository();
+
+            ActDummy act1 = new ActDummy();
+            act1.Active = true;
+            activity.Acts.Add(act1);
+
+            ObservableList<IAct> actionList = activity.Acts;
+            Act action = (Act)actionList[0];
+            action.Active = true;
+
+            mGR.CurrentBusinessFlow.CurrentActivity.Acts.CurrentItem = action;
+
+            //Act
+            mGR.RunAction(action);
+
+            //Assert
+            Assert.AreEqual(action.Status, Amdocs.Ginger.CoreNET.Execution.eRunStatus.Passed);
+            Assert.AreEqual(mGR.CurrentBusinessFlow.CurrentActivity.Acts.CurrentItem,actionList[1]);
+        }
+      
+        public Activity GetActivityFromRepository()
+        {
+            Context context = new Context();
+
+            ObservableList<BusinessFlow> bfList = SR.GetAllRepositoryItems<BusinessFlow>();
+            BusinessFlow BF1 = bfList[0];
+
+            ObservableList<Activity> activityList = BF1.Activities;
+            Activity activity = activityList[0];
+            activity.Active = false;
+
+            context.BusinessFlow = BF1;
+            context.Activity = activity;
+            mGR.CurrentBusinessFlow = BF1;
+            mGR.CurrentBusinessFlow.CurrentActivity = activity;
+            mGR.Context = context;
+
+            ProjEnvironment environment = new ProjEnvironment();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename 'environment' which hides the field with the same name.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4817688)